### ISSUE TITLE
fconv.py, about `decoder_padding_mask`

### ISSUE THREAD

**Issue by ruidongtd** [pytorch/fairseq#652]
In the file `fcony.py`，`class AttentionLayer(nn.Module)`， `encoder_padding_mask` is used for `src`，why don't have `decoder_padding_mask` for `target `?

**Comment by ruidongtd** [comment]
@Myle Ott

**Comment by ruidongtd** [comment]
@myleott

**Comment by myleott** [comment]
When performing attention we need to mask padding symbols that appear on the source (encoder) side that we don’t attend to them. In the decoder, the padding appears on the right side of the sequence, so it doesn’t affect computation. There are some subtle details about this for fconv. Please check out this really nice tutorial about this written by a community member: http://www.telesens.co/2019/04/21/understanding-incremental-decoding-in-fairseq/

**Comment by ruidongtd** [comment]
thank you
